packages/datagateway-common/src/views/selectionAlert.component.tsx

```diff
@@ -0,0 +1,168 @@
+        marginTop: '8px',
+        marginLeft: ({ marginSide }) =>
+          marginSide === undefined ? '0px' : marginSide,
+        marginRight: ({ marginSide }) =>
+          marginSide === undefined ? '0px' : marginSide,
+        paddingTop: '2px',
+        paddingBottom: '2px',
+      },
+      animate: {
+        backgroundColor: 'orange',
+        color: 'white',
+        width: ({ width }) => (width === undefined ? 'auto' : `${width}`),
+        marginTop: '8px',
+        marginLeft: ({ marginSide }) =>
+          marginSide === undefined ? '0px' : marginSide,
+        marginRight: ({ marginSide }) =>
+          marginSide === undefined ? '0px' : marginSide,
+        paddingTop: '2px',
+        paddingBottom: '2px',
+        animation: `$pulsate 700ms ${theme.transitions.easing.easeInOut}`,
+      },
+      '@keyframes pulsate': {
+        '0%': {
+          backgroundColor: 'orange',
+        },
+        '25%': {
+          backgroundColor: 'red',
+        },
+        '50%': {
+          backgroundColor: 'orange',
+        },
+        '75%': {
+          backgroundColor: 'red',
+        },
+        '100%': {
+          backgroundColor: 'orange',
+        },
+      },
+      text: {
+        textAlign: 'center',
+        justifyItems: 'center',
+        alignItems: 'center',
+        flex: '1',
+      },
+    })
+);
+
+const SelectionAlert = React.memo(
+  (props: {
+    selectedItems: DownloadCartItem[];
+    navigateToSelections: () => void;
+    width?: string;
+    marginSide?: string;
+  }): React.ReactElement => {
+    const classes = selectionAlertStyles({
+      width: props.width,
+      marginSide: props.marginSide,
+    });
+    const [t] = useTranslation();
+    const [alertOpen, setAlertOpen] = React.useState(false);
+    const [animating, setAnimating] = React.useState(false);
+    const [alertText, setAlertText] = React.useState('');
+    //Store number of selections on last update of this component (for keeping track of changes to cart)
+    const [numSelectedItems, setNumSelectedItems] = React.useState(0);
+
+    //Current number of selections
+    const newNumSelecItems = props.selectedItems.length;
+
+    //Check for a change and assign text based on increase or decrease
+    if (newNumSelecItems !== numSelectedItems) {
+      const difference = newNumSelecItems - numSelectedItems;
+
+      //Obtain plural if needed
+      const itemText =
+        Math.abs(difference) > 1
+          ? t('selec_alert.items')
+          : t('selec_alert.item');
+
+      if (difference > 0)
+        setAlertText(`
+            ${difference} ${itemText} ${t('selec_alert.added')}`);
+      else
+        setAlertText(`
+            ${difference * -1} ${itemText} ${t('selec_alert.removed')}`);
+
+      setNumSelectedItems(newNumSelecItems);
+      //Change has occurred so need to ensure displayed
+      setAlertOpen(true);
+      setAnimating(true);
+    }
+
+    return (
+      <React.Fragment>
```

Comment:
Why do we have a Fragment here? We only normally need these if we're returning multiple React components but here we're only returning a singular one.

Reply:
@louise-davies This is because the return cannot be null and the contents is conditional. I am not sure if there is a better way other than returning some other empty element.

Reply:
@louise-davies I managed to sort this out, it was because I had explicitly stated the function should return ReactElement.